### PR TITLE
route float messages to float route

### LINKED ISSUE #387: Core: [route float] does not route floats
In Pd this will also route messages that are just float objects.

In Heavy we create switch/case logic that only checks hashes.

Maybe we can add this as a special case.

### PR DESCRIPTION
**Pull request by dromer**
Closes #387 

Needs some more testing and edge case detection.

### CODE REVIEW

File: hvcc/generators/ir2c/ControlSwitchcase.py [hunk 1]
```diff
@@ -67,8 +67,27 @@ def get_C_impl(
...
 
-        out_list.append("switch (msg_getHash(m, msgIndex)) {")
         cases = objects[obj_id].args["cases"]
+
+        # special float case, if present route float messages
+        if objects[obj_id].args.get("is_route", False) and "float" in cases:
+            for i, c in enumerate(cases):
+                if c == "float":
+                    out_list.append("if (msg_getNumElements(m) == 1 && msg_getType(m, 0) == HV_MSG_FLOAT) {")
+                    out_list.append("HvMessage *n = HV_MESSAGE_ON_STACK(2);")
```
[the remark this answers is not in the document]
Comment (dromer) [reply hvcc/generators/ir2c/ControlSwitchcase.py]: This is not how route works.
